src/Coordination/Changelog.cpp

```diff
@@ -824,24 +1056,33 @@ void Changelog::applyEntriesFromBuffer(uint64_t index, nuraft::buffer & buffer)
     }
 }
 
-void Changelog::flush()
+bool Changelog::flush()
 {
-    if (flushAsync())
+    if (auto failed_ptr = flushAsync())
     {
         std::unique_lock lock{durable_idx_mutex};
-        durable_idx_cv.wait(lock, [&] { return last_durable_idx == max_log_id; });
+        durable_idx_cv.wait(lock, [&] { return *failed_ptr || last_durable_idx == max_log_id; });
+
+        return !*failed_ptr;
     }
+
+    return true;
```

Comment:
Shouldn't it return false if did not flush the log due to shutdown? However, I did not find any usages of the returned value.

Reply:
I'll add a comment, NuRaft internally will abort if flush fails (abort is the only way of handling errors) but the config append can happen after changelog was already shut down so we basically don't care about the flush anymore (we want to stop processing things) so I think it's okay.
Also, this flush is called only for config changes and not for other requests.